segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/WriteQueue.java

```diff
@@ -169,9 +168,11 @@ synchronized void add(Write write) {
      * Removes all the completed writes (whether successful or failed) from the beginning of the queue, until the first
      * non-completed item is encountered or the queue is empty.
      *
-     * @return True if there are items left in the queue, false otherwise.
+     * @return A CleanupStatus representing the state of the Operation. If there were failed writes, this will be WriteFailed,
+     * otherwise it will be one of QueueEmpty or QueueNotEmpty, depending on the final state of the queue when this method
+     * finishes.
      */
-    synchronized EnumSet<CleanupStatus> removeFinishedWrites() {
+    synchronized CleanupStatus removeFinishedWrites() {
```

Comment:
The change to this class is returning a status rather than an enum set of statuses. I tried to understand from the PR description whether this is the change that is preventing the spurious exceptions. If this is the case, could you elaborate on how this is being achieved? Otherwise, what's the purpose of the change.

Reply:
1. I tried to solve the problem by doing adding an early `return` statement from the method logging that message 
2. Checkstyle started complaining that there were too many `return` statements in that method.
3. I had to refactor the method to have fewer return statements.
4. During this I either had to make the code using this method a lot more uglier (another nested `if`) or refactor this method to return a single Enum (it really, only returned one useful value to begin with anyway) and keep the original method still look decent.